GPU/GLES/DepthBufferGLES.cpp

```diff
@@ -0,0 +1,183 @@
+// Copyright (c) 2012- PPSSPP Project.
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, version 2.0 or later versions.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License 2.0 for more details.
+
+// A copy of the GPL 2.0 should have been included with the program.
+// If not, see http://www.gnu.org/licenses/
+
+// Official git repository and contact information can be found at
+// https://github.com/hrydgard/ppsspp and http://www.ppsspp.org/.
+
+#include <algorithm>
+#include "gfx_es2/gpu_features.h"
+#include "Core/ConfigValues.h"
+#include "Core/Reporting.h"
+#include "GPU/Common/GPUStateUtils.h"
+#include "GPU/GLES/DrawEngineGLES.h"
+#include "GPU/GLES/FramebufferManagerGLES.h"
+#include "GPU/GLES/ShaderManagerGLES.h"
+#include "GPU/GLES/TextureCacheGLES.h"
+
+static const char *depth_dl_fs = R"(
+#ifdef GL_ES
+precision highp float;
+#endif
+#if __VERSION__ >= 130
+#define varying in
+#define texture2D texture
+#define gl_FragColor fragColor0
+out vec4 fragColor0;
+#endif
+varying vec2 v_texcoord0;
+uniform float u_depthScaleFactor;
+uniform sampler2D tex;
+void main() {
+  float depth = texture2D(tex, v_texcoord0).r;
+  float offset = 0.5 * (u_depthScaleFactor - 1.0) * (1.0 / u_depthScaleFactor);
```

Comment:
here you can just divide by u_depthScaleFactor, computing the inverse within parenthesis doesn't gain us anything

Reply:
Yeah, I think I just copied this from c++ for now, was trying to get it work first before optimizing or simplifying really.  Could probably pass offset as a separate uniform to make sure the shader is simple.

-[Unknown]